src/Sylius/Bundle/PaymentBundle/Doctrine/ORM/PaymentRequestRepository.php

```diff
@@ -61,4 +61,20 @@ public function findOneByHash(string $hash): ?PaymentRequestInterface
             ->getOneOrNullResult()
         ;
     }
+
+    /**
+     * @param array<string> $states
+     * @return array<PaymentRequestInterface>
+     */
+    public function findByStatesAndPaymentId(array $states, int|string $paymentId): array
```

Comment:
Since all usages of this method are based of single state, is `$states` instead of `$state` necessary?